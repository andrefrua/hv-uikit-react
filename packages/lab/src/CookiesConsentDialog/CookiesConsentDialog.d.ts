import * as React from "react";
import { StandardProps } from "@material-ui/core";
import { HvDialogProps } from "@hv/uikit-react-core";

export type HvCookiesConsentDialogClassKey =
  | "root"
  | "closeButton";

export interface HvCookiesConsentDialogProps
  extends StandardProps<HvDialogProps, HvCookiesConsentDialogClassKey> {
  /**
   * The title of the dialog
   */
  title: string;
  /**
   * The content of the dialog
   */
  description: React.ReactNode | string;
  /**
   * The buttons of the dialog
   */
  buttons: React.ReactNode;
}

export default function HvCookiesConsentDialog(
  props: HvCookiesConsentDialogProps
): JSX.Element | null;
