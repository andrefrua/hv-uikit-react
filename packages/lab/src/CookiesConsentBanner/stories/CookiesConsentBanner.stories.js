import React from "react";

import { HvButton, HvTypography } from "@hv/uikit-react-core";

import { HvCookiesConsentBanner } from "../..";

export default {
  title: "Lab/CookiesConsentBanner",
  parameters: {
    componentSubtitle: null,
    usage: "import { HvCookiesConsentBanner } from '@hv/uikit-react-lab'",
  },
  component: HvCookiesConsentBanner,
};

export const Main = () => {
  const title = "This website uses cookies";
  const description = (
    <>
      This website uses cookies to offer you a better browsing experience. You consent to our
      cookies by clicking &quot;Accept&quot;. For more information, visit&nbsp;
      <HvTypography variant="link" component="a" href="http://www.hitachi.com">
        Hitachi Cookie Policy
      </HvTypography>
      .
    </>
  );
  const buttons = (
    <>
      <HvButton category="ghost" style={{ width: 70 }}>
        Accept
      </HvButton>
      <HvButton category="ghost" style={{ width: 150 }}>
        See cookie settings
      </HvButton>
    </>
  );

  return <HvCookiesConsentBanner title={title} description={description} buttons={buttons} />;
};
