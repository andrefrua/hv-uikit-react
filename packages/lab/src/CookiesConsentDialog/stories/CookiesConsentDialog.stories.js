import React from "react";

import { HvButton, HvTypography } from "@hv/uikit-react-core";

import { HvCookiesConsentDialog } from "../..";

export default {
  title: "Lab/CookiesConsentDialog",
  parameters: {
    componentSubtitle: null,
    usage: "import { HvCookiesConsentDialog } from '@hv/uikit-react-lab'",
  },
  component: HvCookiesConsentDialog,
};

export const Main = () => {
  const [open, setOpen] = React.useState(false);

  const title = "This website uses cookies";
  const description = (
    <>
      This website uses cookies to offer you a better browsing experience. You consent to our
      cookies by clicking &quot;Accept&quot;. For more information, visit&nbsp;
      <HvTypography variant="link" component="a" href="http://www.hitachi.com">
        Hitachi Cookie Policy
      </HvTypography>
      .
    </>
  );
  const buttons = (
    <>
      <HvButton category="ghost">Accept</HvButton>
      <HvButton category="ghost">See cookie settings</HvButton>
    </>
  );

  return (
    <>
      <HvButton onClick={() => setOpen(true)}>Cookies Dialog</HvButton>
      <HvCookiesConsentDialog
        title={title}
        description={description}
        buttons={buttons}
        open={open}
        onClose={() => setOpen(false)}
      />
    </>
  );
};
