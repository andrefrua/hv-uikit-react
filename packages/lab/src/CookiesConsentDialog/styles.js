const styles = () => ({
  root: {},
  closeButton: {
    display: "none",
  },
});

export default styles;
