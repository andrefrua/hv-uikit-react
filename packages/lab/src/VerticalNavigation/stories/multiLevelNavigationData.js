import {
  BarChart,
  Open,
  Deploy,
  Binoculars,
  Cpu,
  Machine,
  LightOff,
  Magnet,
  MarginsOff,
  Connect,
  CloseAll,
  ColorPicker,
  DaemonSet,
} from "@hitachivantara/uikit-react-icons";

const multiLevelNavigationData = [
  {
    id: "menu1",
    label: "Menu 1",
    path: "",
    icon: <Open />,
    parent: null,
    data: [
      {
        id: "subMenu1-1",
        label: "Sub Menu 1-1",
        path: "",
        icon: <Binoculars />,
        parent: null,
      },
      {
        id: "subMenu1-2",
        label: "Sub Menu 1-2",
        path: "",
        icon: <Cpu />,
        parent: null,
        data: [
          {
            id: "subMenu1-2-1",
            label: "Sub Menu 1-2-1",
            path: "",
            icon: <LightOff />,
            parent: null,
          },
          {
            id: "subMenu1-2-2",
            label: "Sub Menu 1-2-2",
            path: "",
            icon: <Magnet />,
            parent: null,
          },
          {
            id: "subMenu1-2-3",
            label: "Sub Menu 1-2-3",
            path: "",
            icon: <MarginsOff />,
            parent: null,
          },
        ],
      },
      {
        id: "subMenu1-3",
        label: "Sub Menu 1-3",
        path: "",
        icon: <Machine />,
        parent: null,
      },
    ],
  },
  {
    id: "menu2",
    label: "Menu 2",
    path: "",
    icon: <BarChart />,
    parent: null,
  },
  {
    id: "menu3",
    label: "Menu 3",
    path: "",
    icon: <Deploy />,
    parent: null,
  },
  {
    id: "menu4",
    label: "Menu 4",
    path: "",
    icon: <Connect />,
    parent: null,
    data: [
      {
        id: "menu4-1",
        label: "Menu 4-1",
        path: "",
        icon: <ColorPicker />,
        parent: null,
      },
      {
        id: "menu4-2",
        label: "Menu 4-2",
        path: "",
        icon: <DaemonSet />,
        parent: null,
      },
    ],
  },
  {
    id: "menu5",
    label: "Menu 5",
    path: "",
    icon: <CloseAll />,
    parent: null,
  },
];

export default multiLevelNavigationData;
