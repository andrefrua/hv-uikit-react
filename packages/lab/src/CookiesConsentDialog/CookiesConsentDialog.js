import React from "react";
import PropTypes from "prop-types";
import clsx from "clsx";

import { withStyles } from "@material-ui/core";

import { HvDialog, HvDialogTitle, HvDialogContent, HvDialogActions } from "@hv/uikit-react-core";

import styles from "./styles";

const CookiesConsentDialog = (props) => {
  const { className, classes, title, description, buttons, open, onClose, ...others } = props;

  return (
    <HvDialog
      className={clsx(className, classes.root)}
      classes={{ closeButton: classes.closeButton }}
      open={open}
      onClose={onClose}
      {...others}
    >
      <HvDialogTitle>{title}</HvDialogTitle>
      <HvDialogContent>{description}</HvDialogContent>
      <HvDialogActions>{buttons}</HvDialogActions>
    </HvDialog>
  );
};

CookiesConsentDialog.propTypes = {
  /**
   * Class names to be applied.
   */
  className: PropTypes.string,
  /**
   * A Jss Object used to override or extend the styles applied.
   */
  classes: PropTypes.shape({
    /**
     * Styles applied to the component root class.
     */
    root: PropTypes.string,
    /**
     * Styles applied to the dialog's close button.
     */
    closeButton: PropTypes.string,
  }).isRequired,
  /**
   * The title of the dialog
   */
  title: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
  /**
   * The content of the dialog
   */
  description: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
  /**
   * The buttons of the dialog
   */
  buttons: PropTypes.node.isRequired,
  /**
   * If `true` the dialog is open, if `false`dialog is close
   */
  open: PropTypes.bool,
  /**
   * The functions that is triggered when the dialog closes.
   */
  onClose: PropTypes.func,
};

export default withStyles(styles, { name: "CookiesConsentDialog" })(CookiesConsentDialog);
