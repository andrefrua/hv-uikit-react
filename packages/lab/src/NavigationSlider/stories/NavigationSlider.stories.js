import { useState } from "react";
import useMediaQuery from "@mui/material/useMediaQuery";
import { useTheme } from "@mui/styles";

import {
  HvHeader,
  HvButton,
  HvHeaderActions,
  HvHeaderNavigation,
} from "@hitachivantara/uikit-react-core";
import { BarChart, Open, User, Deploy, Menu } from "@hitachivantara/uikit-react-icons";

import HvNavigationSlider from "../NavigationSlider";

const headerNavigationData = [
  {
    id: "1",
    label: "Overview",
  },
  {
    id: "2",
    label: "Events",
  },

  {
    id: "3",
    label: "Asset",
  },
];

const verticalNavigationData = [
  {
    id: "menu1",
    label: "Menu 1",
    path: "",
    icon: <Open />,
    data: [
      {
        id: "menu1-1",
        label: "Menu 1-1",
        path: "",
        icon: <Open />,
        parent: null,
      },
      {
        id: "menu1-2",
        label: "Menu 1-2",
        path: "",
        icon: <BarChart />,
        data: [
          {
            id: "menu1-2-1",
            label: "Menu 1-2-1",
            path: "",
            icon: <Open />,
            parent: null,
          },
          {
            id: "menu1-2-2",
            label: "Menu 1-2-2",
            path: "",
            icon: <BarChart />,
            parent: null,
          },
          {
            id: "menu1-2-3",
            label: "Menu 1-2-3",
            path: "",
            icon: <Deploy />,
            parent: null,
          },
        ],
        parent: null,
      },
      {
        id: "menu1-3",
        label: "Menu 1-3",
        path: "",
        icon: <Deploy />,
        parent: null,
      },
    ],
    parent: null,
  },
  {
    id: "menu2",
    label: "Menu 2",
    path: "",
    icon: <BarChart />,
    parent: null,
  },
  {
    id: "menu3",
    label: "Menu 3",
    path: "",
    icon: <Deploy />,
    parent: null,
  },
];

export default {
  title: "Lab/Navigation Slider",
  parameters: {
    componentSubtitle: null,
    usage: 'import { HvNavigationSlider } from "@hitachivantara/uikit-react-lab"',
  },
  component: HvNavigationSlider,
  decorators: [(storyFn) => <div style={{ height: "600px" }}>{storyFn()}</div>],
};

export const Main = () => {
  const [selectedItem, setSelectedItem] = useState("menu1-3");

  const navigationChangeHandler = (event, item) => {
    setSelectedItem(item.id);
  };

  return (
    <HvNavigationSlider
      title="Main Example"
      data={verticalNavigationData}
      selected={selectedItem}
      topPosition={0}
      onNavigationChange={navigationChangeHandler}
    />
  );
};

export const ExampleWithDefaultHeader = () => {
  const [selectedItem, setSelectedItem] = useState("menu1");

  const navigationChangeHandler = (event, item) => {
    setSelectedItem(item.id);
  };

  const Header = () => (
    <HvHeader position="fixed">
      <HvHeaderNavigation data={headerNavigationData} />
      <HvHeaderActions>
        <HvButton icon>
          <User />
        </HvButton>
      </HvHeaderActions>
    </HvHeader>
  );

  return (
    <>
      <Header />
      <HvNavigationSlider
        data={verticalNavigationData}
        selected={selectedItem}
        onNavigationChange={navigationChangeHandler}
      />
    </>
  );
};

export const ExampleWithCustomHeader = () => {
  const headerHeight = 80;
  const [selectedItem, setSelectedItem] = useState("menu1");

  const navigationChangeHandler = (event, item) => {
    setSelectedItem(item.id);
  };

  const Header = () => {
    const styles = {
      width: "100%",
      left: 0,
      top: 0,
      position: "fixed",
      display: "flex",
      justifyContent: "center",
      alignItems: "center",
      height: `${headerHeight}px`,
      backgroundColor: "lightblue",
    };
    return <header style={styles}>This is a custom header</header>;
  };

  return (
    <>
      <Header />
      <HvNavigationSlider
        topPosition={headerHeight}
        data={verticalNavigationData}
        selected={selectedItem}
        onNavigationChange={navigationChangeHandler}
      />
    </>
  );
};

export const ExampleWithToggle = () => {
  const [selectedItem, setSelectedItem] = useState("menu1");
  const [isPanelOpen, setIsPanelOpen] = useState(false);

  const navigationChangeHandler = (event, item) => {
    setSelectedItem(item.id);
  };

  const Header = () => {
    return (
      <HvHeader position="fixed">
        <HvButton
          style={{ width: 32, height: 32 }}
          icon
          onClick={() => setIsPanelOpen((prevState) => !prevState)}
        >
          <Menu />
        </HvButton>

        <HvHeaderNavigation data={headerNavigationData} />
        <HvHeaderActions>
          <HvButton icon>
            <User />
          </HvButton>
        </HvHeaderActions>
      </HvHeader>
    );
  };

  return (
    <>
      <Header />
      {isPanelOpen && (
        <HvNavigationSlider
          data={verticalNavigationData}
          selected={selectedItem}
          onNavigationChange={navigationChangeHandler}
        />
      )}
    </>
  );
};

export const ExampleResponsive = () => {
  const theme = useTheme();
  const [selectedItem, setSelectedItem] = useState("1");
  const isMdUp = useMediaQuery(theme.breakpoints.up("md"));
  const [isPanelOpen, setIsPanelOpen] = useState(false);

  const navigationChangeHandler = (event, item) => {
    setSelectedItem(item.id);
  };

  const Header = () => {
    return (
      <HvHeader position="fixed">
        {!isMdUp && (
          <HvButton
            style={{ width: 32, height: 32 }}
            icon
            onClick={() => setIsPanelOpen((prevState) => !prevState)}
          >
            <Menu />
          </HvButton>
        )}
        {isMdUp && <HvHeaderNavigation data={headerNavigationData} />}
        <HvHeaderActions>
          <HvButton icon>
            <User />
          </HvButton>
        </HvHeaderActions>
      </HvHeader>
    );
  };

  return (
    <>
      <Header />
      {isPanelOpen && !isMdUp && (
        <HvNavigationSlider
          data={headerNavigationData}
          selected={selectedItem}
          onNavigationChange={navigationChangeHandler}
        />
      )}
    </>
  );
};
