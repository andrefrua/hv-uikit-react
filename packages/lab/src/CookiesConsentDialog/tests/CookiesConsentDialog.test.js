/* eslint-env jest */

import React from "react";

import { render } from "testing-utils";

import { HvCookiesConsentDialog } from "../..";

import { Main } from "../stories/CookiesConsentDialog.stories";

describe("CookiesConsentDialog", () => {
  describe("sample snapshot testing", () => {
    it("Main", () => {
      const { container } = render(<Main />);
      expect(container).toMatchSnapshot();
    });
  });

  describe("general", () => {
    it("renders the component as expected", () => {
      const { getByRole } = render(<HvCookiesConsentDialog open />);

      const container = getByRole("dialog");

      expect(container).toBeInTheDocument();
    });
  });
});
