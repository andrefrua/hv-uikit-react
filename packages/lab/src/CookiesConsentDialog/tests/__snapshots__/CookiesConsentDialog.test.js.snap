// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CookiesConsentDialog sample snapshot testing Main 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root HvButton-root MuiButton-contained HvButton-primarySVG MuiButton-containedPrimary HvButton-primary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Cookies Dialog
    </span>
  </button>
</div>
`;
